### DISCUSSION TITLE
Either-or modeling

### DISCUSSION THREAD

**Discussion by ASL-rmarshall** [cdisc-org/analysis-results-standard#140] [Ideas]
There are several places in the model where it would be beneficial to specify that an attribute can be one of a choice of classes.  Examples include:
- An item in an ordered list of outputs can be either a sublist or an output
- A where clause can be either a simple condition (dataset/variable/comparator/value) or a compound expression
- A compared value in a where clause condition can be either a single value or a multi-value list

At the moment, the first two examples are implemented simply as two different attributes, only one of which is expected to be used.  However, there is currently nothing in the model to constrain the use of the attributes.  The third example is currently implement as a multi-value list, which nothing in the model to constrain the number of values depending on the comparator.

Need to determine whether modeling of these types of constraint is needed and, if so, how should it be implemented.

**Comment by drewcdisc** [comment]
Need to move this into a Discussion, as this PBI cannot be completed independently.

**Comment by drewcdisc** [comment]
@ASL-rmarshall @bhavinbusa I converted issue #17 into Discussion #140
